vme/src/dilfld.cpp

```diff
@@ -2352,10 +2358,11 @@ void dilfe_fld(dilprg *p)
 
             if (v->type == DILV_UINT4R)
             {
-                if ((v1->val.ptr) && IS_OBJ((unit_data *)v1->val.ptr) && (v2->val.num >= 0) && (v2->val.num < 5))
+                auto *object = reinterpret_cast<obj_data *>(v1->val.ptr);
+                if (object && IS_OBJ(object) && (v2->val.num >= 0 && v2->val.num < static_cast<sbit32>(object->getValueArraySize())))
```

Comment:
I'm curious why you can't simply write (sbit32) on a simple type instead of the longer static_cast... ? :)

Reply:
c-style casts shouldn't be used in C++ code. Things shouldn't be cast, they should be the right type in the first place. Plus c++ casts are easier to find - grep for `sbit32` returns a lot more then grepping for `static_cast<sbit32>` 

I also think c++ casts were designed to be unwieldy  so that it makes you think of what you are doing - gone are the days of everything is a void * :-)